### PR TITLE
https://jira.desk.technology/browse/MSDKI-17

### PR DESCRIPTION
**Pull request by eldark88**
@nartiles please review

### CODE REVIEW

File: DeskKit/DeskKit/View Models/DKContactUsViewModel.m [hunk 1]
```diff
@@ -291,9 +291,11 @@ - (NSDictionary *)caseDictionary
 - (NSDictionary *)messageDictionary
 {
     NSString *fromEmail = [self bestFromEmail];
+    NSString *body = self.bodyItem.text.string.dkTrimmedString;
+    
     NSMutableDictionary *dictionary = [@{
                                          DKMessageDirectionKey: @"in",
-                                         DKMessageBodyKey: self.bodyItem.text.string,
```
Comment (nartiles) [inline DeskKit/DeskKit/View Models/DKContactUsViewModel.m]: Why are we trimming the body?

Comment (eldark88) [inline DeskKit/DeskKit/View Models/DKContactUsViewModel.m]: I don't see a reason to not to trim. If we don't trim it on client side, API will trim it anyway. That's the behavior I notice.

Comment (nartiles) [inline DeskKit/DeskKit/View Models/DKContactUsViewModel.m]: If it's not adding any benefit, why spend the extra method calls?

Comment (eldark88) [inline DeskKit/DeskKit/View Models/DKContactUsViewModel.m]: I think it's a client's responsibility to send accurate data to the server. Also, user can enter a lot of empty spaces and send it to server. Each empty space is counted as a character and increases a payload. By truncating leading and trailing spaces, we are sending more accurate data and removing extra payload.

Comment (nartiles) [inline DeskKit/DeskKit/View Models/DKContactUsViewModel.m]: I disagree with the notion that modifying the users content in this case is making the data more accurate. On the contrary, what if the user wants white spaces for a reason, maybe ascii art, who knows. I can certainly imagine other situation where trimming whitespace changes the meaning of what the user intended. Trimming these characters is modifying what the user wanted to send. And as to the question of which is more efficient, we would need measurements of the cost of sending extra characters vs more cpu time on the extra method calls.

**Comment by eldark88** [comment]
@nartiles please review

**Comment by nartiles** [comment]
🚢

**Pull request merged**